core/controllers/reader_test.py

```diff
@@ -182,6 +182,35 @@ def test_give_feedback_handler(self):
         self.logout()
 
 
+class ExplorationHandlerTest(test_utils.GenericTestBase):
+    """Test the handler for providing the initial data
+    for a single exploration.
+    """
+
+    def test_error_if_exploration_is_none(self):
+        """Test that a missing exploration raises
+        a page not found exception.
+        """
+
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+        self.login(self.VIEWER_EMAIL)
+        exp_id = '6'
+        exp_services.delete_demo(exp_id)
+        exp_services.load_demo(exp_id)
+
+        def _mock_none_function(*_, **__):
+            """Mocks None."""
+            return None
+
+        with self.swap(
+            exp_fetchers, 'get_exploration_by_id', _mock_none_function):
```

Comment:
Can you use `self.swap_with_checks` to make sure `get_exploration_by_id` is called the way you expect? Also, `_mock_none_function` isn't mocking `None`, it's mocking `get_exploration_by_id`, so please name it accordingly